reactioncommerce/reaction
# GraphQL: create reconcileCarts mutation

Resolves #4360   
Impact: **minor**  
Type: **feature**

## Changes
Implements the `reconcileCarts` GraphQL mutation, which was already in the schema.

## Breaking changes
None

## Testing
1. Call `reconcileCarts` as an account with no cart. Verify that the anonymous cart is converted to an account cart and destroyed.
1. Call `reconcileCarts` as an account with a cart, specifying `mode: "merge"`. Verify that items from the two carts are merged together into the account cart, and the anonymous cart is destroyed. An item that is in both carts should have the two quantities summed during merge.
1. Call `reconcileCarts` as an account with a cart, specifying `mode: "keepAnonymousCart"`. Verify that all anonymous cart items are now in the account cart, all items that were in the account cart are gone, and the anonymous cart is destroyed.
1. Call `reconcileCarts` as an account with a cart, specifying `mode: "merge"`. Verify that all account cart items remain in the account cart and none of the anonymous cart items were added to the account cart. Verify that the  Verify that all anonymous cart items are now in the account cart, all items that were in the account cart are gone, and the anonymous cart is destroyed.
